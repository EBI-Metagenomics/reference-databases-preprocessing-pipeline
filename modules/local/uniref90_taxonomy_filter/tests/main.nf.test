nextflow_process {

    name "Test Process UNIREF90_TAXONOMY_FILTER"
    script "../main.nf"
    process "UNIREF90_TAXONOMY_FILTER"

    tag "uniref90"

    test("Run UNIREF90_TAXONOMY_FILTER on uncompressed fasta") {

        when {
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/uniref90.test.fasta", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

        test("Run UNIREF90_TAXONOMY_FILTER on fasta.gz") {

        when {
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/uniref90.test.fasta.gz", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }


}
