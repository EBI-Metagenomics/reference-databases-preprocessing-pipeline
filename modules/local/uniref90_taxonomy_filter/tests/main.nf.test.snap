{
    "Run UNIREF90_TAXONOMY_FILTER on fasta.gz": {
        "content": [
            {
                "0": [
                    "uniref90.filtered.fasta:md5,e5ee4d75d12e9dd95fd7f7ffe9f32208"
                ],
                "1": [
                    "uniref90.protid2taxid:md5,c05d080f20acf5d54afdcd0c3c40463d"
                ],
                "2": [
                    "versions.yml:md5,cb46d8e9c6c847ee8b5a15bf7e4f96d8"
                ],
                "filtered_proteins": [
                    "uniref90.filtered.fasta:md5,e5ee4d75d12e9dd95fd7f7ffe9f32208"
                ],
                "protid2taxid": [
                    "uniref90.protid2taxid:md5,c05d080f20acf5d54afdcd0c3c40463d"
                ],
                "versions": [
                    "versions.yml:md5,cb46d8e9c6c847ee8b5a15bf7e4f96d8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-03T15:06:22.645545"
    },
    "Run UNIREF90_TAXONOMY_FILTER on uncompressed fasta": {
        "content": [
            {
                "0": [
                    "uniref90.filtered.fasta:md5,e5ee4d75d12e9dd95fd7f7ffe9f32208"
                ],
                "1": [
                    "uniref90.protid2taxid:md5,c05d080f20acf5d54afdcd0c3c40463d"
                ],
                "2": [
                    "versions.yml:md5,cb46d8e9c6c847ee8b5a15bf7e4f96d8"
                ],
                "filtered_proteins": [
                    "uniref90.filtered.fasta:md5,e5ee4d75d12e9dd95fd7f7ffe9f32208"
                ],
                "protid2taxid": [
                    "uniref90.protid2taxid:md5,c05d080f20acf5d54afdcd0c3c40463d"
                ],
                "versions": [
                    "versions.yml:md5,cb46d8e9c6c847ee8b5a15bf7e4f96d8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-03T15:06:18.276799"
    }
}